### ISSUE TITLE
 Duckdns was updated currectly but logs falsely show update failed

### ISSUE THREAD

**Issue by Christophe999s** [ddclient/ddclient#503]
ddclient on OPNsense 23.1 updates duckdns correctly, but logs shows it has failed:
```
2023-01-27T20:18:08	Notice	ddclient[56504]	62219 - [meta sequenceId="24"] FAILED: updating myhost: Server said: '0'	
2023-01-27T20:18:08	Notice	ddclient[56504]	61917 - [meta sequenceId="23"] RECEIVE:	
2023-01-27T20:18:08	Notice	ddclient[56504]	61630 - [meta sequenceId="22"] RECEIVE: 0	
2023-01-27T20:18:08	Notice	ddclient[56504]	61396 - [meta sequenceId="21"] RECEIVE: OK	
2023-01-27T20:18:08	Notice	ddclient[56504]	61106 - [meta sequenceId="20"] RECEIVE: 2	
2023-01-27T20:18:08	Notice	ddclient[56504]	60884 - [meta sequenceId="19"] RECEIVE:	
2023-01-27T20:18:08	Notice	ddclient[56504]	60687 - [meta sequenceId="18"] RECEIVE: X-Frame-Options: DENY	
2023-01-27T20:18:08	Notice	ddclient[56504]	60576 - [meta sequenceId="17"] RECEIVE: X-Clacks-Overhead: GNU Terry Pratchett	
2023-01-27T20:18:08	Notice	ddclient[56504]	60319 - [meta sequenceId="16"] RECEIVE: Server: nginx/1.20.0	
2023-01-27T20:18:08	Notice	ddclient[56504]	59994 - [meta sequenceId="15"] RECEIVE: Set-Cookie: AWSALBCORS=xxxxxx/xxxxx; Expires=Fri, 03 Feb 2023 19:18:07 GMT; Path=/; SameSite=None; Secure	
2023-01-27T20:18:08	Notice	ddclient[56504]	59698 - [meta sequenceId="14"] RECEIVE: Set-Cookie: AWSALB=xxxxx/xxxxxx; Expires=Fri, 03 Feb 2023 19:18:07 GMT; Path=/	
2023-01-27T20:18:08	Notice	ddclient[56504]	59587 - [meta sequenceId="13"] RECEIVE: Connection: close	
2023-01-27T20:18:08	Notice	ddclient[56504]	59336 - [meta sequenceId="12"] RECEIVE: Transfer-Encoding: chunked	
2023-01-27T20:18:07	Notice	ddclient[56504]	59172 - [meta sequenceId="11"] RECEIVE: Date: Fri, 27 Jan 2023 19:18:07 GMT	
2023-01-27T20:18:07	Notice	ddclient[56504]	58911 - [meta sequenceId="10"] RECEIVE: HTTP/1.1 200 OK	
2023-01-27T20:18:07	Notice	ddclient[56504]	58684 - [meta sequenceId="9"] SENDING:	
2023-01-27T20:18:07	Notice	ddclient[56504]	58684 - [meta sequenceId="8"] SENDING: Connection: close	
2023-01-27T20:18:07	Notice	ddclient[56504]	58684 - [meta sequenceId="7"] SENDING: User-Agent: ddclient/3.10.0	
2023-01-27T20:18:07	Notice	ddclient[56504]	58684 - [meta sequenceId="6"] SENDING: Host: www.duckdns.org	
2023-01-27T20:18:07	Notice	ddclient[56504]	58684 - [meta sequenceId="5"] SENDING: GET /update?domains=myhost&token=mytoken&ip=xxx.xxx.xxx.xxx HTTP/1.1	
2023-01-27T20:18:07	Notice	ddclient[56504]	58334 - [meta sequenceId="4"] CONNECTED: using SSL	
2023-01-27T20:18:07	Notice	ddclient[56504]	57718 - [meta sequenceId="3"] CONNECT: www.duckdns.org	
2023-01-27T20:18:07	Notice	ddclient[56504]	57417 - [meta sequenceId="2"] UPDATE: updating myhost	
2023-01-27T20:18:07	Notice	ddclient[56504]	57050 - [meta sequenceId="1"] INFO: setting IP address to xxx.xxx.xxx.xxx for myhost
```

/usr/local/etc/ddclient.conf:
```
syslog=yes                  # log update msgs to syslog
pid=/var/run/ddclient.pid   # record PID in file.
verbose=yes
ssl=yes


use=if, if=pppoe0, \
protocol=duckdns, \
password=myduckdnstoken \
myhost

use=if, if=pppoe0, \
protocol=cloudflare, \
zone=mydomain.be, \
login=token, \
password=mycftoken \
sub.mydomain.be

```

/var/tmp/ddclient.cache:
```
## ddclient-3.10.0
## last updated at Fri Jan 27 20:20:59 2023 (1674847259)
atime=0,host=myhost,mtime=0,status=failed,status-ipv4=noconnect,status-ipv6=noconnect,warned-min-error-interval=1674847259,warned-min-interval=0,wtime=30 myhost
atime=1674847259,backupmx=0,host=sub.mydomain.be,ipv4=xxx.xxx.xxx.xxx,mtime=1674847259,mx=,static=0,status=noconnect,status-ipv4=good,status-ipv6=noconnect,warned-min-error-interval=0,warned-min-interval=0,wildcard=0,wtime=0 sub.mydomain.be
```

**Comment by lesterlo** [comment]
I have the same problem

**Comment by drinn** [comment]
Proposed fix at #506

**Comment by raditp** [comment]
Is it possible to release a bugfix version for this issue? It's currently affecting lots of opnsense users who use ddclient version 3.10.
